src/Native/System.Native/pal_io.cpp

```diff
@@ -851,5 +872,39 @@ extern "C" int32_t ReadStdinUnbuffered(void* buffer, int32_t bufferSize)
 #endif
 }
 
+extern "C" int32_t INotifyInit()
+{
+#if HAVE_INOTIFY_INIT
+    return inotify_init();
+#else
+    errno = ENOTSUP;
+    return -1;
+#endif
+}
+
+extern "C" int32_t INotifyAddWatch(int32_t fd, const char* pathName, uint32_t mask)
+{
+    assert(fd >= 0);
+    assert(pathName != nullptr);
+
+#if HAVE_INOTIFY_ADD_WATCH
+    return inotify_add_watch(fd, pathName, mask);
+#else
+    (void)mask;
```

Comment:
Looks like some more of these are needed here and in INotifyRemoveWatch to suppress unused parameter warnings:
```
/Users/dotnet-bot/j/workspace/dotnet_corefx_mac_nativecomp_release_prtest/src/Native/System.Native/pal_io.cpp:885:44: error: unused parameter 'fd' [-Werror,-Wunused-parameter]
extern "C" int32_t INotifyAddWatch(int32_t fd, const char* pathName, uint32_t mask)
                                           ^
/Users/dotnet-bot/j/workspace/dotnet_corefx_mac_nativecomp_release_prtest/src/Native/System.Native/pal_io.cpp:885:60: error: unused parameter 'pathName' [-Werror,-Wunused-parameter]
extern "C" int32_t INotifyAddWatch(int32_t fd, const char* pathName, uint32_t mask)
                                                           ^
/Users/dotnet-bot/j/workspace/dotnet_corefx_mac_nativecomp_release_prtest/src/Native/System.Native/pal_io.cpp:899:47: error: unused parameter 'fd' [-Werror,-Wunused-parameter]
extern "C" int32_t INotifyRemoveWatch(int32_t fd, int32_t wd)
                                              ^
/Users/dotnet-bot/j/workspace/dotnet_corefx_mac_nativecomp_release_prtest/src/Native/System.Native/pal_io.cpp:899:59: error: unused parameter 'wd' [-Werror,-Wunused-parameter]
extern "C" int32_t INotifyRemoveWatch(int32_t fd, int32_t wd)
```